lib/fluent/plugin/buffer/chunk.rb

```diff
@@ -141,19 +144,40 @@ def purge
           self
         end
 
-        def read
+        def read(**kwargs)
           raise NotImplementedError, "Implement this method in child class"
         end
 
         def open(&block)
           raise NotImplementedError, "Implement this method in child class"
         end
 
-        def write_to(io)
+        def write_to(io, **kwargs)
           open do |i|
             IO.copy_stream(i, io)
           end
         end
+
+        module Decompressable
+          include Fluent::Plugin::Compressable
+
+          def read(**kwargs)
+            return super if kwargs[:compress] == :gzip
```

Comment:
This keyword argument name should be `compressed`, not `compress`.
Because caller of this method expects "compressed data", not expecting "compress data here (again)".